kobo/hub/xmlrpc/worker.py

```diff
@@ -227,7 +229,7 @@ def get_tasks_to_assign(request):
 def get_awaited_tasks(request, awaited_task_list):
     task_list = []
     for task in Task.objects.filter(awaited=True, parent__in=[ i["id"] for i in awaited_task_list ]):#.order_by("-exclusive", "-awaited", "id")[:50]:
-        task_info = task.export()
+        task_info = task.export(False)
```

Comment:
Could you please explain this change?  It's passing flat=False, i.e. requesting extra fields in the response.

get_awaited_tasks is called by the workers, if they've never needed those extra fields before then why do they need them now?

Reply:
The reason is that `take_task` crash if the task is not exclusive. You can look into `take_task` and you will see that it requires extra fields from arch and channel, if flat=True these extra fields are not provided.

I'm not sure how this works in production, my guess is all tasks are exclusive, this way will never go into that code.

`get_awaited_tasks` is callend only by `get_next_task` and this calls `take_task`, since I wrote many test cases and all are working this does not affect TaskManager, but I'm not sure if could affect other place.

[Code of take_task](https://github.com/release-engineering/kobo/blob/master/kobo/worker/taskmanager.py#L304)

Reply:
It's not the case that all tasks are exclusive.  I think the reason this has not been detected before is that get_awaited_tasks is only called if the worker is locked, which is a part of gracefully shutting down a worker - probably a rarely-used feature (definitely not used by our team).  By my reading the crash would only happen if you ask a worker to shut down and it still has at least one task in progress.

It's a valuable bug fix but it shouldn't be hidden in a commit with a message implying that it only adds autotests.  Would you be willing to file an issue for it, and have the issue mentioned in your commit message?

Reply:
No problem, I can create an issue for that.